src/v/security/config_rcl.cc

```diff
@@ -183,5 +187,74 @@ result<parsed_url> parse_url(std::string_view url_view) {
     return result;
 }
 
+static constexpr std::string_view mapping_rule_pattern
+  = R"(\/((\\.|[^\\/])*)\/((\\.|[^\\/])*)\/([LU]?).*?|(.*?))";
+
+constexpr std::optional<std::string_view> make_sv(const std::csub_match& sm) {
+    return sm.matched
+             ? std::string_view{sm.first, static_cast<size_t>(sm.length())}
+             : std::optional<std::string_view>{std::nullopt};
+}
+
+result<principal_mapping_rule>
+parse_principal_mapping_rule(std::string_view mapping) {
+    if (!mapping.starts_with("$.")) {
+        return errc::invalid_principal_mapping;
+    }
+    auto slash = mapping.find('/');
+    auto path = ss::sstring(mapping.substr(1, slash - 1));
+    std::replace(path.begin(), path.end(), '.', '/');
+
+    auto pointer = json::Pointer{path};
+    if (!pointer.IsValid()) {
+        return errc::invalid_principal_mapping;
+    }
+
+    std::regex rule_parser{
+      mapping_rule_pattern.begin(),
+      mapping_rule_pattern.length(),
+      std::regex_constants::ECMAScript | std::regex_constants::optimize};
+
+    tls::rule rule;
+    if (slash != std::string_view::npos) {
+        auto rule_str = mapping.substr(slash);
+        std::cmatch components_match;
+        if (!std::regex_search(
+              rule_str.begin(),
+              rule_str.end(),
+              components_match,
+              rule_parser,
+              std::regex_constants::match_default)) {
+            return errc::invalid_principal_mapping;
+        }
+        if (components_match.prefix().matched) {
+            return errc::invalid_principal_mapping;
+        }
+        if (components_match.suffix().matched) {
+            return errc::invalid_principal_mapping;
+        }
+        if (!components_match[1].matched) {
+            return errc::invalid_principal_mapping;
+        }
```

Comment:
im not seeing caess for this in the coverage report. are the in the unit tests?

Reply:
To be fair most of this was copied from the mtls_identity code, I just tweaked the regex slightly and the match offsets. The unit tests don't check this code path. I'll add tests.

Is the coverage report available in the build, or did you run it manually?

Reply:
i just ran it manually to take a peak. i've been doing that locally trying to figure out if it is robust. basically this

```
cmake --preset coverage
ninja <test>
./<test>
../../tools/gcovr
```

sometimes i have to hack the flags to gcovr to get it to work im not yet sure what a constant set of flags is.

Reply:
oh just to reiterate here, no need to actually chase the coverage goal unless you are curious. the tooling isn't really easy to use yet i've just been testing it out on different prs :)